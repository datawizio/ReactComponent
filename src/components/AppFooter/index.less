@import "../../styles/mixin/retina-image";

.ant-layout-footer.main-footer {
  border-top: 1px solid #e9e9e9;
  padding: 13px;
  text-align: center;

  .retina-image(
    "/static/images/nielsen-logo.png",
    "/static/images/nielsen-logo@2x.png",
    85px,
    30px
  );
  background-repeat: no-repeat;
  background-position: calc(100% - 12px);
  padding-right: 104px;
  @media (min-width: 700px) {
    padding-right: 13px;
  }
}
