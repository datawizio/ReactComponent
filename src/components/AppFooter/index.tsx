import React, { useMemo, useContext } from "react";
import { Layout as AntLayout } from "antd";
import ConfigContext from "../ConfigProvider/context";
import "./index.less";

interface AppFooterProps {
  text?: string;
}

const AppFooter: React.FC<AppFooterProps> = ({ text }) => {
  const year = useMemo(() => new Date().getFullYear(), []);
  const { translate } = useContext(ConfigContext);

  return (
    <AntLayout.Footer className="main-footer">
      {text || translate("COPYRIGHT", { year })}
    </AntLayout.Footer>
  );
};

export default AppFooter;
