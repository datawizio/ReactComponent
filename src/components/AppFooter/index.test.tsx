import "jsdom-global/register";
import React from "react";
import { mount } from "enzyme";

import AppFooter from "./index";

const mockProps = {};

const setUp = (props?) => mount(<AppFooter {...props} />);

describe("AppFooter component", () => {
  let component;
  beforeEach(() => {
    component = setUp(mockProps);
  });

  it("Render AppFooter correctly", () => {
    expect(component).toMatchSnapshot();
  });

  it("AppFooter has correct class", () => {
    expect(component.find(".main-footer").length).toBeTruthy();
  });

  it("AppFooter has correct translate key", () => {
    expect(component.html()).toContain("COPYRIGHT");
  });
});
