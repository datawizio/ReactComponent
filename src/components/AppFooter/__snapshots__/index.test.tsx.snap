// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter component Render AppFooter correctly 1`] = `
<AppFooter>
  <Footer
    className="main-footer"
  >
    <Basic
      className="main-footer"
      prefixCls="ant-layout-footer"
      tagName="footer"
    >
      <footer
        className="ant-layout-footer main-footer"
      >
        COPYRIGHT
      </footer>
    </Basic>
  </Footer>
</AppFooter>
`;
